### DISCUSSION TITLE
How to run a shell-less docker image?

### DISCUSSION THREAD

**Discussion by chriseaton** [concourse/concourse#6665] [Help & Support] [answered]
We are building a custom automation that utilizes a headless gcr.io/distroless/nodejs image. Problem is, we're unsure how to actually run the container. We tried using a task, but it requires a `run` `path` which causes the headless container to fail - the image already has a CMD defined... how can we simply just "run the container"? Is that even possible?

Trying to use a task results in:
`Backend error: Exit status: 500, message: {"Type":"","Message":"runc exec: exit status 1: exec failed: container_linux.go:367: starting container process caused: exec: \"/app/index.js\": permission denied","Handle":"","ProcessID":"","Binary":""}`

**Comment by aoldershaw** [comment, marked as answer]
I don't think there's a way to simply inherit the `CMD` of a docker container. Looks like the error is specifying `run.path` to `/app/index.js`, which isn't an executable - instead, I think it should be something like:

```yaml
run:
  path: node
  args: [/app/index.js]
```

**Reply by chriseaton** [reply]
Unfortunately tried that, and get:

> Backend error: Exit status: 500, message: {"Type":"ExecutableNotFoundError","Message":"exec failed: container_linux.go:367: starting container process caused: exec: \"node\": executable file not found in $PATH","Handle":"","ProcessID":"","Binary":""}

I did find this interesting note on their documentation, but I'm unsure how to emulate "vector format" in concourse, I tried quoting, single quoting, single-quoting double quotes...

![image](https://user-images.githubusercontent.com/5247568/111260888-d522e580-85de-11eb-9920-48cf7577dbbc.png)

**Reply by aoldershaw** [reply]
Looking at the image, it appears `node` isn't on the `PATH` (as the error indicates), but there's still the node binary (at `/nodejs/bin/node` which you can run, e.g.:

```yaml
run:
  path: /nodejs/bin/node
  args: [/app/index.js]
```

**Reply by chriseaton** [reply]
That did it thanks!
